engine/tester/fixtures/tx_fixtures.go

```diff
@@ -0,0 +1,272 @@
+	WithOPReturn(dataStr string) GivenTXSpec
+	WithOutputScript(parts ...ScriptPart) GivenTXSpec
+	WithP2PKHOutput(satoshis uint64) GivenTXSpec
+
+	TX() *trx.Transaction
+	InputUTXO(inputID int) bsv.Outpoint
+	InputSourceTX(inputID int) *trx.Transaction
+	ID() string
+	BEEF() string
+	RawTX() string
+	EF() string
+}
+
+type txSpec struct {
+	utxos          []*trx.UTXO
+	outputs        []*trx.TransactionOutput
+	t              testing.TB
+	disableSigning bool
+
+	grandparentTXIndex int
+	sourceTransactions map[string]*trx.Transaction
+}
+
+// GivenTX creates a new GivenTXSpec for building a MOCK! transaction
+func GivenTX(t testing.TB) GivenTXSpec {
+	return &txSpec{
+		t:                  t,
+		grandparentTXIndex: 0,
+		sourceTransactions: make(map[string]*trx.Transaction),
+	}
+}
+
+// WithoutSigning disables signing of the transaction (default is false)
+func (spec *txSpec) WithoutSigning() GivenTXSpec {
+	spec.disableSigning = true
+	return spec
+}
+
+// WithInput adds an input to the transaction with the specified satoshis
+// it automatically creates a parent tx (sourceTX) with P2PKH UTXO with provided satoshis
+func (spec *txSpec) WithInput(satoshis uint64) GivenTXSpec {
+	return spec.WithSingleSourceInputs(satoshis)
+}
+
+// WithSingleSourceInputs adds inputs to the transaction with the specified satoshis
+// All the inputs will be sourced from a single parent transaction
+func (spec *txSpec) WithSingleSourceInputs(satoshis ...uint64) GivenTXSpec {
+	sourceTX := spec.makeParentTX(satoshis...)
+	for i, s := range satoshis {
+		i32, _ := conv.IntToUint32(i)
+		utxo, err := trx.NewUTXO(sourceTX.TxID().String(), i32, P2PKHLockingScript.String(), s)
+		require.NoErrorf(spec.t, err, "creating utxo for input: %d", i)
+
+		utxo.UnlockingScriptTemplate = P2PKHUnlockingScriptTemplate
+
+		spec.utxos = append(spec.utxos, utxo)
+	}
+	spec.sourceTransactions[sourceTX.TxID().String()] = sourceTX
+
+	return spec
+}
+
+// WithOPReturn adds an OP_RETURN output to the transaction with the specified data
+func (spec *txSpec) WithOPReturn(dataStr string) GivenTXSpec {
+	data := []byte(dataStr)
+	o, err := trx.CreateOpReturnOutput([][]byte{data})
+	require.NoError(spec.t, err, "creating op return output")
+
+	spec.outputs = append(spec.outputs, o)
+
+	return spec
+}
+
+// WithP2PKHOutput adds a P2PKH output to the transaction with the specified satoshis
+func (spec *txSpec) WithP2PKHOutput(satoshis uint64) GivenTXSpec {
+	spec.outputs = append(spec.outputs, &trx.TransactionOutput{
+		Satoshis:      satoshis,
+		LockingScript: P2PKHLockingScript,
+	})
+	return spec
+}
+
+// ScriptPart is an interface for building script parts
+type ScriptPart interface {
+	Append(s *script.Script)
+}
+
+// OpCode is an alias for byte to represent an opcode and implements ScriptPart for script building
+type OpCode byte
+
+// Append appends the opcode to the script
+func (op OpCode) Append(s *script.Script) {
+	_ = s.AppendOpcodes(script.OpRETURN)
```

Comment:
maybe at least panic on error?

Reply:
In `go-sdk` they ignore the error for this (for some reason), so I didn't want to be "more saint that the pope" 🙈

Reply:
Done